tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,335 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier4
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pvc import get_all_pvcs, delete_pvcs
+from ocs_ci.ocs.resources.pod import (
+    get_mds_pods, get_mon_pods, get_mgr_pods, get_osd_pods, get_all_pods,
+    get_fio_rw_iops
+)
+from ocs_ci.utility.utils import TimeoutSampler, ceph_health_check
+from tests.helpers import (
+    verify_volume_deleted_in_backend, wait_for_resource_state,
+    wait_for_resource_count_change, verify_pv_mounted_on_node
+)
+from tests import disruption_helpers
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@pytest.mark.parametrize(
+    argnames=['interface', 'resource_to_delete'],
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'mgr'],
+            marks=pytest.mark.polarion_id("OCS-810")
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'mon'],
+            marks=pytest.mark.polarion_id("OCS-811")
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, 'osd'],
+            marks=pytest.mark.polarion_id("OCS-812")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mgr'],
+            marks=pytest.mark.polarion_id("OCS-813")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mon'],
+            marks=pytest.mark.polarion_id("OCS-814")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'osd'],
+            marks=pytest.mark.polarion_id("OCS-815")
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, 'mds'],
+            marks=pytest.mark.polarion_id("OCS-816")
+        )
+    ]
+)
+class TestResourceDeletionDuringMultipleDeleteOperations(ManageTest):
+    """
+    Delete ceph/rook pod while deletion of PVCs, pods and IO are progressing
+    """
+    num_of_pvcs = 25
+    pvc_size = 3
+
+    @pytest.fixture()
+    def setup_base(
+        self, interface, multi_pvc_factory, pod_factory
+    ):
+        """
+        Create PVCs and pods
+        """
+        pvc_objs = multi_pvc_factory(
+            interface=interface,
+            project=None,
+            storageclass=None,
+            size=self.pvc_size,
+            access_mode=constants.ACCESS_MODE_RWO,
```

Comment:
please cover RWX mode for this scenario. If that is done in any other script then please let me know that.

Reply:
Creating PVCs of different access modes will be enabled by modifying multi_pvc_factory to choose access modes randomly or by distributing the given access mode options equally among the PVCs.
Working on this requirement separately through https://github.com/red-hat-storage/ocs-ci/issues/382#issuecomment-518155983